cuda_core/tests/test_device.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021-2024 NVIDIA Corporation.  All rights reserved.
+#
+# Please refer to the NVIDIA end user license agreement (EULA) associated
+# with this source code for terms and conditions that govern your use of
+# this software. Any use, reproduction, disclosure, or distribution of
+# this software and related documentation outside the terms of the EULA
+# is strictly prohibited.
+
+from cuda import cuda, cudart
+from cuda.core.experimental._device import Device
+from cuda.core.experimental._utils import handle_return, ComputeCapability, CUDAError, \
+                             precondition
+import pytest
+
+@pytest.fixture(scope='module')
+def init_cuda():
+    Device().set_current()
+
+def test_device_initialization():
+    device = Device()
+    assert device is not None

```

Comment:
I think for this test we should check (by not passing in any fixture) initializing a `Device` object does not initialize the CUDA context, perhaps via `cuCtxGetCurrent`.